### PR TITLE
Show durations in GitHub statuses

### PR DESCRIPTION
**Pull request by bduffany**
image

---

**Version bump**: Patch 

**Related issues**: Fixes https://github.com/buildbuddy-io/buildbuddy-internal/issues/675

### CODE REVIEW

File: server/util/timeutil/timeutil.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package timeutil
```
Comment (tylerwilliams) [inline server/util/timeutil/timeutil.go:1]: no good deed goes unpunished.... can you send a followup CL that uses this library for all the conversions we do to and from time.Time from usec?

File: server/build_event_protocol/build_status_reporter/build_status_reporter.go [hunk 2]
```diff
@@ -198,13 +200,37 @@ func (r *BuildStatusReporter) githubPayloadFromTestSummaryEvent(event *build_eve
 
 func (r *BuildStatusReporter) githubPayloadFromFinishedEvent(event *build_event_stream.BuildEvent) *github.GithubStatusPayload {
 	description := descriptionFromExitCodeName(event.GetFinished().ExitCode.Name)
```
Comment (siggisim) [inline server/build_event_protocol/build_status_reporter/build_status_reporter.go:202]: Can we make this something a bit more user friendly than `OK in 32s` maybe something like `Succeeded in 32s` or `Passed in 32s` or `Success in 32s`

**Review by siggisim — APPROVED** [review]

File: server/util/timeutil/timeutil.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import (
+	"time"
+)
+
+// UnixMillis converts a Unix timestamp in milliseconds to a `time.Time`.
+func UnixMillis(timestamp int64) time.Time {
+	sec := timestamp / 1e3
+	nsec := (timestamp - sec*1e3) * 1e6
+	return time.Unix(sec, nsec)
+}
```
Comment (tylerwilliams) [inline server/util/timeutil/timeutil.go:12]: how bout:

func FromMillis(timestampMillis int64) time.Time {
	return time.Unix(0, timestampMillis * 1e6)
}

File: server/build_event_protocol/build_status_reporter/build_status_reporter.go [hunk 4]
```diff
@@ -198,13 +200,37 @@ func (r *BuildStatusReporter) githubPayloadFromTestSummaryEvent(event *build_eve
...
+	}
+	if d >= 1*time.Second {
+		return fmt.Sprintf("%ds", d/time.Second)
+	}
+	if d >= 1*time.Millisecond {
+		return fmt.Sprintf("%dms", d/time.Millisecond)
+	}
+	return "< 1ms"
+}
+
```
Comment (tylerwilliams) [inline server/build_event_protocol/build_status_reporter/build_status_reporter.go:233]: nit: move this to timeutil

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/util/timeutil/timeutil.go (hunk 3 above)
Comment (bduffany) [reply server/util/timeutil/timeutil.go:12]: Done

File: server/build_event_protocol/build_status_reporter/build_status_reporter.go (hunk 4 above)
Comment (bduffany) [reply server/build_event_protocol/build_status_reporter/build_status_reporter.go]: Done

File: server/util/timeutil/timeutil.go (hunk 1 above)
Comment (bduffany) [reply server/util/timeutil/timeutil.go:1]: lol, will do

File: server/build_event_protocol/build_status_reporter/build_status_reporter.go (hunk 2 above)
Comment (bduffany) [reply server/build_event_protocol/build_status_reporter/build_status_reporter.go:202]: Went with "Successful" which is what GH does for actions

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**